kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+typedef struct pthread_sem {
+	struct k_sem *sem;
+} sem_t;
+
+int sem_destroy(sem_t *semaphore)
+{
+	k_sem_reset(semaphore);
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	int count;
+
+	count = k_sem_count_get(semaphore);
+	*value = count;
+
+	return 0;
+}
+
+int sem_init(sem_t *semaphore, int pshared, unsigned value)
+{
+	if (semaphore->limit = 0){
+		/* limit can not be zero*/
+		return EINVAL;
+	}
+	ARG_UNUSED(pshared);
```

Comment:
Please assert that this value has an expected value. This makes it possible to support different values later; if it's ignored here, future code using this portability layer will build but most likely fail to run at runtime.